packages/app/src/cli/commands/app/function/run.ts

```diff
@@ -42,10 +44,43 @@ export default class FunctionRun extends Command {
       path: flags.path,
       userProvidedConfigName: flags.config,
       callback: async (_app, ourFunction) => {
+        let functionExport = DEFAULT_FUNCTION_EXPORT
+
+        if (flags.export !== undefined) {
+          debug(`Using export ${flags.export} from the --export flag.`)
+          functionExport = flags.export
+        } else if (
+          ourFunction.configuration.targeting !== undefined &&
+          ourFunction.configuration.targeting.length > 0
+        ) {
+          const targeting = ourFunction.configuration.targeting
+
+          if (targeting.length > 1 && isTTY()) {
+            let targets = targeting.map((target) => ({
+              label: target.target,
+              value: target.export || DEFAULT_FUNCTION_EXPORT,
+            }))
+
+            functionExport = await renderAutocompletePrompt({
+              message: `Which target would you like to execute?`,
+              choices: targets,
+            })
+          } else {
+            functionExport = targeting?.[0]?.export || DEFAULT_FUNCTION_EXPORT
+            debug(
+              `Using export '${functionExport}'. Use the --export flag or an interactive terminal to select a different export.`,
+            )
+          }
+        } else {
+          debug(
```

Comment:
These would normal be `outputDebug` -- they'll work with the normal `--verbose` option then